client/src/controllers/index.ts

```diff
@@ -5,6 +5,7 @@ import { ActionController } from './ActionController';
 import { AutosizeController } from './AutosizeController';
 import { BulkController } from './BulkController';
 import { CloneController } from './CloneController';
+import { CondController } from './CondController';
```

Comment:
question: looking at the other controllers, they all seem to be fully named. Is there a reason this is not `ConditionController` ?

Reply:
I could not think of a better name that was also short.

One of the downsides of a longer name is that this name gets repeated for every single attribute.

For example, comparing the two, we get a much longer value in the DOM. This impacts DOM size and also readability.

```
<input data-w-cond-target="enable" data-action="change->w-cond#something keyup->w-cond#other" />
<input data-w-condition-target="enable" data-action="change->w-condition#something keyup->w-condition#something" />
```

We do say that the controller name should be concise here https://docs.wagtail.org/en/stable/contributing/ui_guidelines.html#how-to-build-a-stimulus-controller


However, I am more than happy to change if this 'make it short' hurts the readability of this.

Reply:
`w-relate` (related elements)
`w-deps` (dependant elements)
`w-match` (matched elements)
`w-rules` (rules determine the behaviour of fields) - maybe this could be a better option?


Another idea, we could re-label the match attribute to be data-conditions maybe. But I like to separate the controller targets from the way we set the 'rules'.

Reply:
No, it makes sense especially in the DOM. Could the controller be more verbosed, but the DOM trigger short? Would `w-rule` work? if not `CondController`/`w-cond` will work (sorry if I am nitpicking)

Reply:
Oh, cross-post! `w-rules` sounds like a winner to me at the moment. Before we commit - would we ever get to a position where we'd have a controller for complex rules?

I wonder what other think

Reply:
I envision we would like to solve this issue https://github.com/wagtail/wagtail/issues/2172 and `RulesPanel` seems a suitable name there. Nothing's stopping us from just making this behaviour a kwarg though. 

`FieldPanel(..., rules=[...]` (or something). 

If we do need to build a new controller for something else where rules is useful as a name for this thing we can just use a different name for that new thing. I'm not aware of anything on our existing Stimulus roadmap that would fit this name though.

The only thing close would be a master config like controller, but it's unlikely we will head in that direction anyway.

Reply:
Here's what the HTML would look like. In this example we are setting the `beta` field to be enabled only when you enter `'go'` into the `alpha` field. Contrived but should give a sense of things.

```
<fieldset data-controller="w-rules" data-action="change->w-rules#resolve">
  <input type="text" name="alpha" />
  <input type="text" name="beta"  data-w-rules-target="enable" data-match='{"alpha": "go"}' />
</fieldset>
```

An alternative approach where the `data-match` is instead `data-rule` (singular).

```
<fieldset data-controller="w-rules" data-action="change->w-rules#resolve">
  <input type="text" name="alpha" />
  <input type="text" name="beta"  data-w-rules-target="enable" data-rule='{"alpha": "go"}' />
</fieldset>
```

A future enhancement may allow for per-target type rules, so it would be good to consider this in the context of changing this attribute. Both variants listed below.

```
<fieldset data-controller="w-rules" data-action="change->w-rules#resolve">
  <input type="text" name="alpha" />
  <input type="text" name="beta"  data-w-rules-target="enable show" data-match-enable='{"alpha": "go"}' data-match-show='{"alpha": "blah"}'  />
  <input type="text" name="beta"  data-w-rules-target="enable show" data-rule-enable='{"alpha": "go"}' data-rule-show='{"alpha": "go"}' />
</fieldset>
```

Reply:
No major opinions from the core team. Aside from the eventual Python API that uses these attributes may end up having a more descriptive name. 

I'm still leaning towards rules but will wait for @zerolab or anyone else to confirm.

Reply:
I like rules, fwiw